### DISCUSSION TITLE
How the cycle number is taking into account in PyBaMM Degradation models

### DISCUSSION THREAD

**Discussion by Akila1993** [pybamm-team/PyBaMM#4437] [Q&A] [answered]
Hello everyone and @DrSOKane, 

I understand that the particle cracking model has a fatigue cracking model where The fatigue crack growth model follows Paris’ law and depends on the duration of a cycle, However, I am not very clear how cycle number affects for SEI layer growth model and Li Plating model, is it the rate of change of concentrations or based any other parameters. 

Also for all three degradation models,  if it's only the time dependence,  will it affect the accuracy of the model if the assigned period is very small for a cycle?
 
If anyone can clarify this, it would be really helpful. Thanks.

**Comment by DrSOKane** [comment, marked as answer]
Hi @Akila1993,

PyBaMM does not use cycle number as a parameter in any degradation mechanism. For Paris' Law crack growth, cycle number N is converted into time t using the chain rule:

dl_cr/dt = dl_cr/dN * dN/dt = dl_cr/dN * 1/3600,

where 1C is assumed to be the typical discharge current.

The period can indeed affect the accuracy of the model, or even if it converges at all. This is to do with events. Basically, if you define an experiment such as `["Discharge at 1C until 2.5 V"]`, the behaviour changes at the point 2.5 V is reached. The word **until** implies an event. PyBaMM's default CasADi solver does not support events, and PyBaMM therefore has to check at certain intervals if the event has been hit. These intervals are controlled by the period. The default period is 1 minute, so if the cut-off is hit between 58 minutes and 59 minutes, PyBaMM goes back to that time range and does additional calculations until it finds exactly when the event happened. However, if the period is too long, this process does not converge.

You should set the period depending on how fast you expect conditions to change during the experiment. For a 1C (dis)charge, a period of 1 minute is reasonable. For less aggressive cycles like a C/3 (dis)charge or a rest period, increasing the period to 5 minutes will significantly degrase the computational resources required. For more intense conditions such as a 3C (discharge), 1 minute may be too long, depending on the cell's power capability.

**Reply by Akila1993** [reply]
Dear Dr @DrSOKane , 

Thank you for your details explanation, It is very helpful. 

However, I am not very clear how cycle number affects the **SEI layer growth model** and **Li Plating model**, is it the rate of change of concentrations or based on any other parameters? 

Can you also please suggest some PyBaMM references to refer for this? 

Thank you for your consideration.

**Reply by DrSOKane** [reply]
Cycle number does not affect SEI layer growth or lithium plating.

**Reply by Akila1993** [reply]
Thank you once again, Dr @DrSOKane.

**Reply by rtimms** [reply]
FYI all of these problems:

> The period can indeed affect the accuracy of the model, or even if it converges at all. This is to do with events. Basically, if you define an experiment such as ["Discharge at 1C until 2.5 V"], the behaviour changes at the point 2.5 V is reached. The word until implies an event. PyBaMM's default CasADi solver does not support events, and PyBaMM therefore has to check at certain intervals if the event has been hit. These intervals are controlled by the period. The default period is 1 minute, so if the cut-off is hit between 58 minutes and 59 minutes, PyBaMM goes back to that time range and does additional calculations until it finds exactly when the event happened. However, if the period is too long, this process does not converge.

>You should set the period depending on how fast you expect conditions to change during the experiment. For a 1C (dis)charge, a period of 1 minute is reasonable. For less aggressive cycles like a C/3 (dis)charge or a rest period, increasing the period to 5 minutes will significantly degrase the computational resources required. For more intense conditions such as a 3C (discharge), 1 minute may be too long, depending on the cell's power capability.

are avoided with the new `IDAKLUSolver`. The role of the period is just to set how frequently you want data to be stored in the solution object.

On cycle number - the SEI, plating and other models don't depend explicitly on cycle number. We solve PDEs that describe how the variables (e.g. SEI thickness) evolve with time.
